.github/workflows/build_and_run_chain_simulator_and_execute_system_test.yml

```diff
@@ -35,48 +35,68 @@ jobs:
       MX_CHAIN_TESTING_SUITE_TARGET_BRANCH: ""
 
     steps:
-      - name: Fetch Latest Comment
-        if: github.event_name != 'issue_comment'
+      - name: Determine Target Branches
+        id: target_branch
+        run: |
+          echo "CURRENT_BRANCH=${GITHUB_HEAD_REF:-${GITHUB_REF_NAME}}" >> $GITHUB_ENV
+
+          # Default target branches based on the PR base branch
+          if [[ "${{ github.event.pull_request.base.ref }}" == "main" ]]; then
+            echo "MX_CHAIN_SIMULATOR_TARGET_BRANCH=main" >> $GITHUB_ENV
+            echo "MX_CHAIN_TESTING_SUITE_TARGET_BRANCH=main" >> $GITHUB_ENV
+          elif [[ "${{ github.event.pull_request.base.ref }}" == "master" ]]; then
+            echo "MX_CHAIN_SIMULATOR_TARGET_BRANCH=main" >> $GITHUB_ENV
+            echo "MX_CHAIN_TESTING_SUITE_TARGET_BRANCH=main" >> $GITHUB_ENV
+          else
+            echo "MX_CHAIN_SIMULATOR_TARGET_BRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
+            echo "MX_CHAIN_TESTING_SUITE_TARGET_BRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
+          fi
+
+          # Always set MX_CHAIN_GO_TARGET_BRANCH based on the PR base branch
+          echo "MX_CHAIN_GO_TARGET_BRANCH=${{ github.event.pull_request.base.ref }}" >> $GITHUB_ENV
+
+
+      - name: Fetch and Parse Last Comment for Branches
         uses: actions/github-script@v7
-        id: fetch_comment
+        id: fetch_and_parse_last_comment
         with:
           github-token: ${{ secrets.GITHUB_TOKEN }}
           script: |
+            // Get the latest comment
             const comments = await github.rest.issues.listComments({
               owner: context.repo.owner,
               repo: context.repo.repo,
               issue_number: context.issue.number,
             });
 
-            // Filter for comments containing "Run Tests:"
-            const latestComment = comments.data.reverse().find(comment => comment.body.includes('Run Tests:'));
+            const lastComment = comments.data.pop(); // Get the last comment
+
+            if (lastComment && lastComment.body.includes('Run Tests:')) {
+              const body = lastComment.body.trim();
+              core.setOutput('latest_comment', body);
 
-            if (latestComment) {
-              core.setOutput('latest_comment', latestComment.body);
+              // Parse the branches from the last comment
+              const simulatorBranchMatch = body.match(/mx-chain-simulator-go:\s*(\S+)/);
+              const testingSuiteBranchMatch = body.match(/mx-chain-testing-suite:\s*(\S+)/);
```

Comment:
:see_no_evil: 

GH workflow inputs (pretty visual forms) would have permitted us to not write regex.

GH workflow inputs support default values (of course, without additional conditions as above, but might be sufficient).